main.tf

```diff
@@ -0,0 +1,666 @@
+    }]
+  })
+
+  tags = local.common_tags 
+
+  # The `depends_on` attribute specifies that this IAM role depends on the `aws_lambda_function.CloudGuardAWPSnapshotsUtilsFunction` resource.
+  depends_on = [aws_lambda_function.CloudGuardAWPSnapshotsUtilsFunction]
+}
+
+## CloudGuard AWP Resources ##
+
+// Note: count - Used as condition to create resources based on the scan mode.
+
+# The CloudGuardAWPCrossAccountRolePolicy resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_policy" "CloudGuardAWP" {
+  name        = "CloudGuardAWP"
+  description = "Policy for CloudGuard AWP"
+  tags = local.common_tags
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "ec2:DescribeInstances",
+          "ec2:DescribeSnapshots",
+          "ec2:DescribeRegions",
+          "ec2:DescribeVolumes"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "lambda:InvokeFunction",
+          "lambda:GetFunction",
+          "lambda:GetLayerVersion",
+          "lambda:TagResource",
+          "lambda:ListTags",
+          "lambda:UntagResource",
+          "lambda:UpdateFunctionCode",
+          "lambda:UpdateFunctionConfiguration",
+          "lambda:GetFunctionConfiguration"
+        ]
+        Resource = aws_lambda_function.CloudGuardAWPSnapshotsUtilsFunction.arn
+      },
+      {
+        Effect = "Allow",
+        Action=  "iam:ListRoleTags",
+        Resource = aws_iam_role.CloudGuardAWPCrossAccountRole.arn 
+      },
+      {
+        Effect   = "Allow"
+        Action   = "cloudformation:DescribeStacks"
+        Resource = "arn:${data.aws_partition.current.partition}:cloudformation:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:stack/*"
+      },
+      {
+        Effect   = "Allow"
+        Action   = "s3:GetObject"
+        Resource = "arn:${data.aws_partition.current.partition}:s3:::${local.agentless_bucket_name}/${local.remote_functions_prefix_key}*"
+      }
+    ]
+  })
+}
+
+# The CloudGuardAWPCrossAccountRoleAttachment resource attaches the CloudGuardAWPCrossAccountRolePolicy to the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_role_policy_attachment" "CloudGuardAWPCrossAccountRoleAttachment" {
+  role       = aws_iam_role.CloudGuardAWPCrossAccountRole.name
+  policy_arn = aws_iam_policy.CloudGuardAWP.arn
+}
+# end resources for CloudGuardAWPCrossAccountRole
+
+# Cross account role policy
+# The CloudGuardAWPCrossAccountRolePolicy resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_policy" "CloudGuardAWPCrossAccountRolePolicy_InAccount" {
+  count       = local.scan_mode == "inAccount" ? 1 : 0
+  name        = "CloudGuardAWPCrossAccountRolePolicy_InAccount"
+  description = "Policy for CloudGuard AWP Cross Account Role"
```

Comment:
change from "Policy for CloudGuard AWP Cross Account Role" to "Policy for CloudGuard AWP Cross Account Role - InAccount Mode"